src/network/core_utils.rs

```diff
@@ -25,6 +25,129 @@ pub struct CoreUtils {
     pub networkns: String,
 }
 
+pub struct IPAMAddresses {
+    // ip addresses for netlink
+    pub container_addresses: Vec<ipnet::IpNet>,
+    pub gateway_addresses: Vec<ipnet::IpNet>,
+    pub ipv6_enabled: bool,
+    // result for podman
+    pub net_addresses: Vec<types::NetAddress>,
+    pub nameservers: Vec<IpAddr>,
+}
+
+const IPAM_HOSTLOCAL: &str = "host-local";
+const IPAM_DHCP: &str = "dhcp";
+const IPAM_NONE: &str = "none";
```

Comment:
Not a blocker: Should we move these constants to https://github.com/containers/netavark/blob/main/src/network/constants.rs as they might get used by other parts of code in future.

Reply:
done